src/handlers/diff_header.rs

```diff
@@ -153,7 +155,72 @@ impl<'a> StateMachine<'a> {
         {
             self.painter.emit()?;
             self._handle_diff_header_header_line(self.source == Source::DiffUnified)?;
-            self.handled_diff_header_header_line_file_pair = self.current_file_pair.clone()
+            self.handled_diff_header_header_line_file_pair = self.current_file_pair.clone();
+        }
+        Ok(handled_line)
+    }
+
+    #[inline]
+    fn test_diff_header_file_operation_line(&self) -> bool {
+        (matches!(self.state, State::DiffHeader(_)) || self.source == Source::DiffUnified)
+            && (self.line.starts_with("deleted file mode ")
+                || self.line.starts_with("new file mode "))
+    }
+
+    /// Check for and handle the "deleted file ..."  line.
```

Comment:
Is `FileEvent::Added` used?

Reply:
`Added` and `Removed` are both with very small amount of use. There is the declaration then 2 locations for each:
in `parse_diff_header_line`:
```rust
        line if line.starts_with("new file mode ") => {
            (line[14..].to_string(), FileEvent::Added) // "new file mode ".len()
        }
        line if line.starts_with("deleted file mode ") => {
            (line[18..].to_string(), FileEvent::Removed) // "deleted file mode ".len()
        }
```

Then in `handle_diff_header_file_operation_line`:
```rust
        match file_event {
            FileEvent::Removed => [...],
            FileEvent::Added => [...],
            _ => (),
        }
```
The `Added` is what allow us to solve #128 

---

I hesitated to modify `get_file_change_description_from_file_paths` to use. This code:
```rust
            (minus_file, "/dev/null", _, _) => [...],
            ("/dev/null", plus_file, _, _) =>  [...],
```
would have become something like:
```rust
            (_, _, FileEvent::Removed, _) => [...],
            (_, _, _, FileEvent::Added) =>  [...],
```
And the complete match in which they are should have been refactored. to no longer take into account the filename but only the event

As it was my first contribution, I judged it was a bit too much of refactoring.
But now that I feel a bit more comfortable with your code if you want I can work on it.